src/test/java/test/Test1.java

```diff
@@ -36,7 +38,8 @@ public void includedGroups() {
 
         tng.run();
 
-        assertThat(listener.getSucceedMethodNames()).containsExactly("method1", "method3");
+        List<String> expected = Arrays.asList("method1", "method3");
+        doAssert(listener.getSucceedMethodNames(), expected);
```

Comment:
I'm not sure about the modification of the current tests (except if the order is wrong).
Instead, we should try to keep the previous behavior.
An option could be to introduce a new edge between tests of the same class.